srfi-167.html

```
@@ -304,7 +304,8 @@ <h3 id="okvs-set"><code>(okvs-set! okvs-or-transaction key value)</code></h3>
     <h3 id="okvs-delete"><code>(okvs-delete! okvs-or-transaction key)</code></h3>
 
     <p>Delete the pair associated with <code>KEY</code> bytevector
-        using <code>OKVS-OR-TRANSACTION</code>.</p>
+        using <code>OKVS-OR-TRANSACTION</code>. If there is no such key,
+        no error is raised.</p>
 
```

All OKVS libraries I have tried do not report an error when deleting a key that does not exists. Hence, it would require to "read before delete" which can be costly. I'd prefer to keep the behavior unspecified when there is no such key.

---

I remember having that discussion as the lmdb egg I'm using does report an error when deleting a key that doesn't exist. Does that mean it is fine to do that as the behavior is unspecified?

---

We could be on the safe side by returning `#f` when there is no such key and `#t` when there is. What do you think?

---

Still FDB and sqlite would return nothing. So, I think it is better to let it unspecified.

---

The interface for returning `#f` or `#t` would mean to not ever report errors, so that would put you back to square one. I wouldn't mind it, but make up your mind whether you prefer errors, return values or lack of specification.

One problem I see here is that if an error may be raised, then that could cancel a transaction, so code might behave differently depending on which implementation you pick.

---

It will be best to ignore any error and return nothing.

---

Because compatibility.

---

But that's what the change specifies, errors are not raised. It doesn't say anything about what is returned though.